crates/core/tedge_api/src/store/pending_entity_store.rs

```diff
@@ -93,10 +93,11 @@ impl PendingEntityStore {
         let mut children = vec![];
         if let Some(direct_children) = self.orphans.remove(entity_tid) {
             for child in direct_children {
-                let pending_entity_cache = self.entities.remove(&child).unwrap();
```

Comment:
The purpose of this method is not clear. It's not obvious why this method updates the pending entity store

- neither from the name (`take_cached_child_entities_data`) 
- nor the doc comment (`/// Returns all the children and their children recursively for a given parent`)

Reply:
For cached children, once the missing parent is registered, all those children can be registered with the entity store. That's when we extract/remove them from the pending store using this API and register them with the entity store. For some reason I thought the word `take` would imply that we're **taking** the owned values from it rather than references. May be, `remove` was more appropriate.

Reply:
I will add appropriate doc comments.

Reply:
See: https://github.com/thin-edge/thin-edge.io/pull/3344/commits/74db7e1c3dc37822ecf592f784dc84cc8ce8b48d